nntrainer/tensor/tensor.cpp

```diff
@@ -327,6 +387,17 @@ Tensor Tensor::getBatchSlice(unsigned int offset, unsigned int size) const {
   return getSharedDataTensor(dim_, offset * this->dim.getFeatureLen());
 }
 
+void Tensor::createSharedDataTensor(const Tensor &src, Tensor &dest,
+                                    unsigned int offset) {
+  if (src.data)
+    dest.data = std::shared_ptr<float>(src.data, src.data.get() + offset);
+  else if (!src.src_tensor)
+    dest.src_tensor = std::make_shared<SrcSharedTensor>(&src, offset);
+  else
+    dest.src_tensor = std::make_shared<SrcSharedTensor>(
+      src.src_tensor->tensor(), offset + src.src_tensor->offset());
```

Comment:
So.. if there is src.src_tensor, then we are going to use memory from src.offset() to offset of dest? If it is, then it is not possible to share from src.data?

Reply:
The function first check If `src.data` exists (meaning tensor already has memory allocated). If it is, then we directly make this tensor based on the src tensor.
If `src.data` does not exist (meaning tensor does not memory allocated), and `src.src_tensor` does not exist (meaning the src tensor does not exist on other tensor), then create a SrcTensor around the src.
If `src.src_tensor` exists, then use the `src.src_tensor` to create the required `src_tensor` to avoid recursive dependency.
Some more explanation in https://github.com/nnstreamer/nntrainer/pull/880#discussion_r567550639.

> So.. if there is src.src_tensor, then we are going to use memory from src.offset() to offset of dest? If it is, then it is not possible to share from src.data?

The function first tries to share from `src.data`. If src.data is not yet allocated, then it goes to src_tensor.

Reply:
Could you add some comments when they are used? and if there is the case which src.data and src.src_tensor is true?

Reply:
done